src/client/components/pages/Courses/CourseInstanceTable.tsx

```diff
@@ -182,6 +256,117 @@ const CourseInstanceTable: FunctionComponent<CourseInstanceTableProps> = ({
+                              }))
+                        }
+                        value={isSEASValue}
+                        name="isSEASValue"
+                        id="isSEASValue"
+                        label="The table will be filtered as selected in this Is SEAS dropdown filter"
+                        isLabelVisible={false}
+                        hideError
+                        onChange={
+                          (event:React.ChangeEvent<HTMLInputElement>) => {
+                            setIsSEASValue(event.currentTarget.value);
+                          }
+                        }
+                      />
+                    </TableHeadingCell>
+                  );
+                }
+                const isFall = field.columnGroup
+                  === COURSE_TABLE_COLUMN_GROUP.FALL;
+                if (field.viewColumn === COURSE_TABLE_COLUMN.OFFERED) {
+                  return (
+                    <TableHeadingCell
+                      scope="col"
+                      key={field.key}
+                    >
+                      <Dropdown
+                        options={
+                          [{ value: 'All', label: 'All' }]
+                            .concat(Object.values(OFFERED)
+                              .map((offeredOption):
+                              {value: string; label: string} => {
+                                const offeredDisplayTitle = offeredEnumToString(
+                                  offeredOption
+                                );
+                                return {
+                                  value: offeredOption,
+                                  label: offeredDisplayTitle,
+                                };
+                              }))
+                        }
+                        value={isFall
+                          ? fallOfferedValue
+                          : springOfferedValue}
+                        name={isFall ? 'fallOfferedValue' : 'springOfferedValue'}
+                        id={isFall ? 'fallOfferedValue' : 'springOfferedValue'}
+                        label={isFall
+                          ? 'The table will be filtered as selected in this fall offered dropdown filter'
+                          : 'The table will be filtered as selected in this spring offered dropdown filter'}
+                        isLabelVisible={false}
+                        hideError
+                        onChange={
+                          (event:React.ChangeEvent<HTMLInputElement>) => {
+                            if (isFall) {
+                              setFallOfferedValue(event.currentTarget.value);
+                            } else {
+                              setSpringOfferedValue(event.currentTarget.value);
+                            }
+                          }
+                        }
+                      />
+                    </TableHeadingCell>
+                  );
+                }
```

Comment:
This is sort of the inverse of the note above about pulling the filter state up into the `CoursesPage` -- Here I think we want to psuh these individual filter details down to the `tableFields` file, similar to how in the table body we've pushed the rendering of the different cell contents behind the `getValue` function. 

I think what we'd need to do is go through the `tableFields` array and add something like a `getFilter` function to each entry, where you'd basically pass in the `filters` object and the `genericFilterUpdate` method described above. Then this whole block could be replaced by adding:

```tsx
  return field.getFilter(filters, genericFilterUpdate);
```

in place of the `return null` on line 380, after the `if (field.viewColumn === COURSE_TABLE_COLUMN.ENROLLMENT)` block

The main benefit would be to reduce the complexity of the layout by removing the need for all of these `if` blocks. Almost all of our fields will have filters, and for those that don't we can always just have `getFilter` return `null`. 

The other benefit is that we can potentially break things down further an reuse more of the logic, i.e. having `getFilter` call different functions to format dropdowns vs. format text fields.

Reply:
Thanks for your help with this! There is now a `getFilter` function in `tableFields` that generates a dropdown filter for the corresponding column.